fix(#16232): use a11y params > element properly

jsomsanith opened the pull request:
Issue: #16232

## What I did
The a11y param > element was not used properly. It is supposed to be an element id too serve as scan root, but the code takes it as the element itself.
Furthermore, if no element is passed in params, it queries a `#story-root` element first that comes from nowhere (no ref found in the repo) before taking `#root`.

I removed this unknown `#story-root` and take the `element` param properly as element id.

## How to test

- [ ] Is this testable with Jest or Chromatic screenshots?
- [x] Does this need a new example in the kitchen sink apps?
- [ ] Does this need an update to the documentation?

If your answer is yes to any of these, please make sure to include it in your PR.
Added a story about a11 parameters in official-storybook example

<!--

Everybody: Please submit all PRs to the `next` branch unless they are specific to the current release. Storybook maintainers cherry-pick bug and documentation fixes into the `master` branch as part of the release process, so you shouldn't need to worry about this. For additional guidance: https://storybook.js.org/docs/react/contribute/how-to-contribute

Maintainers: Please tag your pull request with at least one of the following:
`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

-->

jsomsanith committed:
fix(#16232): use a11y params > element properly

kylegach reviewed (commented):
Thanks for this, @jsomsanith! I left a comment that's worth discussing, but it generally looks good!

jsomsanith committed:
Update addons/a11y/src/a11yRunner.ts

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

jsomsanith committed:
fix

jsomsanith commented on addons/a11y/src/a11yRunner.ts:
```diff
@@ -41,13 +36,14 @@ const run = async (storyId: string) => {
 
-      const { element = getElement(), config, options = {} } = input;
+      const { element = 'root', config, options = {} } = input;
+      const htmlElement = document.getElementById(element);
```
Oops, committed a little fast. Fixed this with 
```javascript
const htmlElement = document.querySelector(`#${element}`);
```

jsomsanith commented:
@kylegach thanks for the review
Fixed with your comment :)

kylegach commented on addons/a11y/src/a11yRunner.ts:
```diff
@@ -41,13 +36,14 @@ const run = async (storyId: string) => {
       const axe = await import('axe-core');
 
-      const { element = getElement(), config, options = {} } = input;
+      const { element = 'root', config, options = {} } = input;
+      const htmlElement = document.querySelector(`#${element}`);
```
Sorry, perhaps I wasn't clear enough in my comment. Here's what I was after:

```suggestion
      const { element = '#root', config, options = {} } = input;
      const htmlElement = document.querySelector(element);
```

This way, consumers can pass, for example, `element: '[data-testid="whatever"]'` instead of being limited to passing an `id`.

jsomsanith replied:
Oh ok, yes I didn't understand.
I was looking to update the addon documentation, but it seems that this element option was meant to be a selector and not an id :D 

[Current documentation](https://storybook.js.org/addons/@storybook/addon-a11y)
<img width="541" alt="image" src="https://user-images.githubusercontent.com/2874505/165828989-5538f4a5-987c-46f5-b946-e4803c49586e.png">

jsomsanith replied:
Suggestion commited, thanks again @kylegach

jsomsanith committed:
Update addons/a11y/src/a11yRunner.ts

Co-authored-by: Kyle Gach <kyle.gach@gmail.com>

kylegach committed:
Update addon-a11y ElementId story

- Updates story's parameters to match changes in previous commit (9feb18832d)
- Minor wording/style changes

kylegach reviewed (approved):
Thanks for this improvement, @jsomsanith! 👏 

I pushed a commit with a fix and some minor wording/style updates. I'll merge as soon as CI does its thing.

kylegach commented:
I don't think the CI failures are related to this PR, so I'm going to merge.

jsomsanith commented:
Thank for the review :)
